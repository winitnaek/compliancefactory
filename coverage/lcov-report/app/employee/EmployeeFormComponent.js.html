<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/employee/EmployeeFormComponent.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/employee</a> EmployeeFormComponent.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import redux from 'redux'
import React, { Component, PropTypes } from 'react'
import { Field, SubmissionError } from 'redux-form'
&nbsp;
export const renderField = field =&gt; {
	const {type, label, input,meta:{touched, error, warning}} = field
	return (
		&lt;div className="form-group"&gt;
		&lt;label&gt;{label}&lt;/label&gt; 
		&lt;br/&gt;
		&lt;input {...input} type={type} className="form-control"/&gt;
		{touched &amp;&amp;
			((<span class="branch-1 cbranch-no" title="branch not covered" >error </span>&amp;&amp; <span class="branch-2 cbranch-no" title="branch not covered" >&lt;span&gt;{error}&lt;/span&gt;)</span> ||
			(<span class="branch-3 cbranch-no" title="branch not covered" >warning </span>&amp;&amp; <span class="branch-4 cbranch-no" title="branch not covered" >&lt;span&gt;{warning}&lt;/span&gt;)</span>)}
		&lt;/div&gt;
	)
}
&nbsp;
class EmployeeFormComponent extends Component {
<span class="fstat-no" title="function not covered" >	my</span>Submit(values) {
<span class="cstat-no" title="statement not covered" >		console.log(values);</span>
<span class="cstat-no" title="statement not covered" >		return this.props.onSave(values).then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; {</span>
<span class="cstat-no" title="statement not covered" >			this.props.reset()</span>
<span class="cstat-no" title="statement not covered" >			return response</span>
		}).catch(<span class="fstat-no" title="function not covered" >er</span>ror =&gt; {
<span class="cstat-no" title="statement not covered" >			throw new SubmissionError(error)</span>
		})
	}
&nbsp;
	render() {
		//const { handleSubmit} = this.props
		return (
			&lt;div className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1"&gt;
            &lt;h1&gt;Add Employee&lt;/h1&gt;
            &lt;form onSubmit={this.props.handleSubmit(<span class="fstat-no" title="function not covered" >(e</span>vent)=&gt;<span class="cstat-no" title="statement not covered" >this.mySubmit(event))</span>}&gt;
                &lt;Field name="first_name" label="First Name" component={renderField} type="text"/&gt;
                &lt;Field name="last_name" label="Last Name" component={renderField} type="text"/&gt;
                &lt;Field name="email" label="email" component={renderField} type="email"/&gt;
                &lt;button type="submit" className="btn btn-primary"&gt;Add Employee&lt;/button&gt;
            &lt;/form&gt;
            &lt;/div&gt;
		)
	}
}
&nbsp;
/*EmployeeFormComponent.propTypes = {
	onSave: PropTypes.func.isRequired,
&nbsp;
	handleSubmit: PropTypes.func.isRequired,
	reset: PropTypes.func.isRequired,
	submitting: PropTypes.bool.isRequired
}*/
&nbsp;
export default EmployeeFormComponent</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 14 2017 17:08:07 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
