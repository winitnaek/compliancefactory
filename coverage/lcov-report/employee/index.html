<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for employee</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> employee
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">25.29% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>43/170</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.07% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/56</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">26.25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>21/80</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.7% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/163</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="EmployeeFormComponent.js"><a href="EmployeeFormComponent.js.html">EmployeeFormComponent.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	</tr>

<tr>
	<td class="file medium" data-value="EmployeeFormContainer.js"><a href="EmployeeFormContainer.js.html">EmployeeFormContainer.js</a></td>
	<td data-value="52.94" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 52%;"></div><div class="cover-empty" style="width:48%;"></div></div></td>
	<td data-value="52.94" class="pct medium">52.94%</td>
	<td data-value="17" class="abs medium">9/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="53.33" class="pct medium">53.33%</td>
	<td data-value="15" class="abs medium">8/15</td>
	</tr>

<tr>
	<td class="file high" data-value="EmployeeListContainer.jsx"><a href="EmployeeListContainer.jsx.html">EmployeeListContainer.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="EmployeeRowContainer.js"><a href="EmployeeRowContainer.js.html">EmployeeRowContainer.js</a></td>
	<td data-value="61.54" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 61%;"></div><div class="cover-empty" style="width:39%;"></div></div></td>
	<td data-value="61.54" class="pct medium">61.54%</td>
	<td data-value="13" class="abs medium">8/13</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="61.54" class="pct medium">61.54%</td>
	<td data-value="13" class="abs medium">8/13</td>
	</tr>

<tr>
	<td class="file low" data-value="employeeAPI.js"><a href="employeeAPI.js.html">employeeAPI.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	</tr>

<tr>
	<td class="file low" data-value="employeeAction.js"><a href="employeeAction.js.html">employeeAction.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="69" class="abs low">0/69</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="69" class="abs low">0/69</td>
	</tr>

<tr>
	<td class="file low" data-value="formValidations.js"><a href="formValidations.js.html">formValidations.js</a></td>
	<td data-value="36.96" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.96" class="pct low">36.96%</td>
	<td data-value="46" class="abs low">17/46</td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="39" class="abs low">3/39</td>
	<td data-value="39.13" class="pct low">39.13%</td>
	<td data-value="23" class="abs low">9/23</td>
	<td data-value="29.27" class="pct low">29.27%</td>
	<td data-value="41" class="abs low">12/41</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 14 2017 10:23:47 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
