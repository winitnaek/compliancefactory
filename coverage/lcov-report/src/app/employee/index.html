<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/app/employee</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/app/employee
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.03% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>87/185</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25.81% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/62</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50.6% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>42/83</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.51% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>81/178</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="EmployeeFormComponent.js"><a href="EmployeeFormComponent.js.html">EmployeeFormComponent.js</a></td>
	<td data-value="40" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="10" class="abs low">4/10</td>
	</tr>

<tr>
	<td class="file medium" data-value="EmployeeFormContainer.js"><a href="EmployeeFormContainer.js.html">EmployeeFormContainer.js</a></td>
	<td data-value="52.94" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 52%;"></div><div class="cover-empty" style="width:48%;"></div></div></td>
	<td data-value="52.94" class="pct medium">52.94%</td>
	<td data-value="17" class="abs medium">9/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="53.33" class="pct medium">53.33%</td>
	<td data-value="15" class="abs medium">8/15</td>
	</tr>

<tr>
	<td class="file high" data-value="EmployeeListContainer.jsx"><a href="EmployeeListContainer.jsx.html">EmployeeListContainer.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file medium" data-value="EmployeeRowContainer.js"><a href="EmployeeRowContainer.js.html">EmployeeRowContainer.js</a></td>
	<td data-value="69.23" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="69.23" class="pct medium">69.23%</td>
	<td data-value="13" class="abs medium">9/13</td>
	</tr>

<tr>
	<td class="file medium" data-value="employeeAPI.js"><a href="employeeAPI.js.html">employeeAPI.js</a></td>
	<td data-value="70" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 70%;"></div><div class="cover-empty" style="width:30%;"></div></div></td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="70" class="pct medium">70%</td>
	<td data-value="10" class="abs medium">7/10</td>
	</tr>

<tr>
	<td class="file low" data-value="employeeAction.js"><a href="employeeAction.js.html">employeeAction.js</a></td>
	<td data-value="29.41" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.41" class="pct low">29.41%</td>
	<td data-value="68" class="abs low">20/68</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="48" class="pct low">48%</td>
	<td data-value="25" class="abs low">12/25</td>
	<td data-value="29.41" class="pct low">29.41%</td>
	<td data-value="68" class="abs low">20/68</td>
	</tr>

<tr>
	<td class="file high" data-value="employeeReducer.js"><a href="employeeReducer.js.html">employeeReducer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	</tr>

<tr>
	<td class="file low" data-value="formValidations.js"><a href="formValidations.js.html">formValidations.js</a></td>
	<td data-value="36.96" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.96" class="pct low">36.96%</td>
	<td data-value="46" class="abs low">17/46</td>
	<td data-value="7.69" class="pct low">7.69%</td>
	<td data-value="39" class="abs low">3/39</td>
	<td data-value="39.13" class="pct low">39.13%</td>
	<td data-value="23" class="abs low">9/23</td>
	<td data-value="29.27" class="pct low">29.27%</td>
	<td data-value="41" class="abs low">12/41</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 14 2017 17:31:11 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
