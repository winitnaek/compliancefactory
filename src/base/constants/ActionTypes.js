export const LOAD_EMPLOYEES='LOAD_EMPLOYEES'
export const LOAD_EMPLOYEES_SUCCESS='LOAD_EMPLOYEES_SUCCESS'
export const CREATE_EMPLOYEE='CREATE_EMPLOYEE'
export const CREATE_EMPLOYEE_SUCCESS='CREATE_EMPLOYEE_SUCCESS'
export const DELETE_EMPLOYEE_SUCCESS='DELETE_EMPLOYEE_SUCCESS'
export const LOAD_MONTHLYCHART_DATA='LOAD_MONTHLYCHART_DATA'
export const LOAD_MONTHLYCHART_DATA_SUCCESS='LOAD_MONTHLYCHART_DATA_SUCCESS'
export const LOAD_IMPORT_SUCCESS='LOAD_IMPORT_SUCCESS'
export const CREATE_IMPORT_SUCCESS='CREATE_IMPORT_SUCCESS'
export const DELETE_IMPORT_SUCCESS='DELETE_IMPORT_SUCCESS'
export const LOAD_APPCONF_SUCCESS='LOAD_APPCONF_SUCCESS'