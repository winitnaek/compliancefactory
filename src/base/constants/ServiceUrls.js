export const EMPLOYEES_URL='employees/'
export const MONTHLYCHART_DATA_URL='monthlyChartdata/'
export const IMPORT_URL='importdata/'

