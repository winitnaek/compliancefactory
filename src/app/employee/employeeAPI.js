import * as svcs from '../../base/constants/ServiceUrls';

class employeeAPI {
  static getAllEmployee(apiUrl) {
   
    var svcs_url = `${apiUrl}${svcs.EMPLOYEES_URL}`;
    return fetch(svcs_url).then(response => {
      return response.json();
    }).catch(error => {
      return error;
    });
  }
  static deleteEmployee(emp,apiUrl) {
    var svcs_url = `${apiUrl}${svcs.EMPLOYEES_URL}${emp.id}`;
    return fetch(svcs_url, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(response => {
      return response.json();
    })
    .catch(error => {
      return error;
    });
  }
  static saveEmployee(values,apiUrl) {
    var svcs_url = `${apiUrl}${svcs.EMPLOYEES_URL}`;
    return fetch(svcs_url, {
      method: 'POST',
      body: JSON.stringify(values),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    .then(response => {
      return response.json();
    })
    .catch(error => {
      return error;
    });
  }
}
export default employeeAPI;