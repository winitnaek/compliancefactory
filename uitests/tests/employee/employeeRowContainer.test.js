import React from 'react';

import EmployeeRow from '../../../src/app/employee/EmployeeRowContainer';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import enzyme,{shallow} from  'enzyme'; 
import shallowWithStore from '../../utils/shallowWithStore.jsx'
import {getAllEmployee} from '../../../src/app/employee/employeeAPI';
import * as svcs from '../../../../garnishment/src/base/constants/ServiceUrls';
import * as types from '../../../../garnishment/src/base/constants/ActionTypes';
import fetchMock from 'fetch-mock';

 const mockStore = configureMockStore();
 
 describe('Employee rows renders correctly', () => {

  let store
  beforeEach(function() {
    store = mockStore({emps:[]});
    enzyme.configure({ adapter: new Adapter() });
  }); 
  it('renders correctly', () => {
    store = mockStore({emps: [
      { "first_name": "Test First Name",
      "last_name": "Test Last Name",
      "email": "testemail@email.com",
      "id": 1}]});
    const component = renderer.create(
      <Provider store={store}>
      <EmployeeRow/>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders Loading to test empty employees', () => {
    const rendered = renderer.create(
      <Provider store={store}>
      <EmployeeRow/>
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot(<tbody>
      <tbody>
        <tr>
          <td>
            Loading..
          </td>
        </tr>
      </tbody>
    </tbody>);
  });
});