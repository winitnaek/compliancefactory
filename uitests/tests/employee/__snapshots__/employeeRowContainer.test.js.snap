// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<tbody>
  <tbody>
    <tr>
      <td>
        Loading..
      </td>
    </tr>
  </tbody>
</tbody>
`;

exports[`Employee rows renders correctly renders correctly 1`] = `
<tbody>
  <tr>
    <td>
      1
    </td>
    <td>
      Test First Name
    </td>
    <td>
      Test Last Name
    </td>
    <td>
      testemail@email.com
    </td>
    <td>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Delete me!
      </button>
    </td>
  </tr>
</tbody>
`;
